### PR TITLE
Nikita/feat: additional fields select

### PR DESCRIPTION
**Pull request by nikita-deriv**

### CODE REVIEW

File: components/FormQR/FormInput.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import styles from "./FormQR.module.scss";
+import QRCodeRender from "../QrCodeRender";
+import AddContext from "../../context/AddContext";
+import { req_fields_number } from "../../utils/values_form_qr";
+
+interface IFormInput {
+    id: string;
+    label: string;
+    maxLength: number;
+
```
Comment (maryiafrantsava-deriv) [inline components/FormQR/FormInput.tsx:15]:
```suggestion
```

File: components/Settings/FieldsSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    useEffect(() => {
+        setCheckedState(additional_fields);
+    }, [additional_fields])
+
+    const handleOnChange = (position: number) => {
+        const updatedCheckedState = checkedState.map((item, index) => index === position ? !item : item);
+
+        setCheckedState(updatedCheckedState);
+        setAdditionalFields(updatedCheckedState);
+
```
Comment (maryiafrantsava-deriv) [inline components/Settings/FieldsSelect.tsx:19]:
```suggestion
```

File: context/AddContext.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+interface IAddContext {
+  additional_fields: Array<boolean>;
+  setAdditionalFields: (fields: Array<boolean>) => void;
+}
+
+const initialValues = {
+  additional_fields : [false, false, false, false, false],
```
Comment (maryiafrantsava-deriv) [inline context/AddContext.tsx]: please, change to mo clean sense as { }

**Review by maryiafrantsava-deriv — CHANGES_REQUESTED** [review]

File: context/AddContext.tsx (hunk 3 above)
Comment (nikita-deriv) [reply context/AddContext.tsx]: This array used for checking checkbox status only. Refactored for using in common cases

**Pull request merged**